h2o-extensions/target-encoder/src/main/java/ai/h2o/targetencoding/TargetEncoder.java

```diff
@@ -235,11 +235,15 @@ Frame imputeNAsForColumn(Frame data, String teColumnName, String strToImpute) {
     
     private void updateDomainGlobally(Frame fr, String teColumnName, String[] domain) {
       Lockable lock = fr.write_lock();
-      Vec updatedVec = fr.vec(teColumnName);
-      updatedVec.setDomain(domain);
-      DKV.put(updatedVec);
-      fr.update();
-      lock.unlock();
+      try {
+        Vec updatedVec = fr.vec(teColumnName);
+        updatedVec.setDomain(domain);
+        fr.update();
+      } finally {
+        lock.unlock();
+        fr.update();
```

Comment:
update after an `unlock`? seems suspicious

example of correct use: https://github.com/h2oai/h2o-3/blob/master/h2o-core/src/main/java/water/fvec/Frame.java#L1832

Reply:
https://github.com/h2oai/h2o-3/pull/4501